misc/package/Data Files/MWSE/core/lib/logger/init.lua

```diff
@@ -0,0 +1,1016 @@
+			(e.g., "herbert100/more quickloot" instad of "more quickloot")
+		2) Some mods might be given names via a corresponding `metadata.toml` file.
+	]]
+
+
+	-- Kill off the part of the path that contains "data files\\mwse\\"
+	local _, mwseFolderEndIndex = filePath:find("data files\\mwse\\", 1, true)
+	if mwseFolderEndIndex then
+		filePath = filePath:sub(mwseFolderEndIndex + 1)
+	end
+	
+	---@type string[]
+	local pathParts = filePath:split("\\")
+
+	-- The root directory. This will be one of "mods", "lib", or "core".
+	local rootDir = table.remove(pathParts, 1)
+
+
+
+	-- now we will initialize the modName and modDir
+	-- we start off by assuming there is no mod author folder, and that the modName and modDir are the same.
+	local modDir = pathParts[1]
+	local modName = modDir
+
+	
+	-- If it's a mod, initialize it using the appropriate runtime.
+	-- Also check if there's a mod author folder.
+	-- By using runtimes, we can determine if a mod has an author folder or not.
+	if rootDir == "mods" then
+		--[[ If there's a mod author folder, then:
+			- `pathParts[1]` is the mod author name
+			- `pathParts[2]` is the `modName`
+			- `pathParts[1].."."..pathParts[2]` is the `modDir`.
+		Otherwise, pathParts[1]` is the `modName` and `modDir`
+		]]
+		local modDirWithAuthorName = pathParts[1].."."..pathParts[2]
+
+		---@type {key: string, path: string, parent_path: string, legacy_mod: boolean, core_mod: boolean, metadata: MWSE.Metadata?}
+		---@diagnostic disable-next-line: undefined-field
+		local runtime = mwse.activeLuaMods[modDirWithAuthorName]
+
+		
+		if runtime then
+			modDir = modDirWithAuthorName
+			modName = pathParts[2]
+		else -- No author directory.
+			---@diagnostic disable-next-line: undefined-field
+			runtime = mwse.activeLuaMods[pathParts[1]]
+		end
+
+		-- Unless we have made an error, the runtime will always exist.
+		-- We'll use it to update the mod name using the mods metadata, if appropriate.
+		if runtime then
+			local pkg = runtime.metadata and runtime.metadata.package
+			modName = pkg and pkg.name or modName
+		else
+			mwse.log(
+				"[Logger | ERROR]: Could not find the runtime information for \"%s\" or \"%s\". \z
+					This should not happen!", 
+				pathParts[1], modDirWithAuthorName
+			)
+		end
+	elseif rootDir == "lib" then
+		-- do nothing
+	elseif rootDir == "core" then
+		-- do nothing
+	else
+		mwse.log(
+			'\t[Logger | ERROR]: Filepath "%s" was not correctly deduced. "%s" is an invalid root direcotry', 
+			filePath, rootDir
+		)
+	end
+	-- This is path of the filepath that does not include the `modDir`.
+	-- NOTE: `modDir` could be either `pathParts[1]` or `pathParts[1] .. "." .. pathParts[2]`.
+	-- Thus, `modDir` is NOT a substring of `table.concat(pathParts, "/")`, 
+	-- as the directory delimters may differ.
+	local relativeFilePath = table.concat(pathParts, "/"):sub(modDir:len() + 2)
+
+	return modName, modDir, relativeFilePath
+end
+
+
+
+-- Logging framework.
+---@class Logger : Logger.SharedData
```

Comment:
Why isn't this class named `mwseLogger`? Isn't the new logger meant to replace the current one?